backends/nxp/backend/ir/converter/node_converters/shared/conv_utils.py

```diff
@@ -0,0 +1,400 @@
+# Copyright 2023-2025 NXP
+#
+# License: LA_OPT_NXP_Software_License
+# See the LICENSE_LA_OPT_NXP_Software_License for more details.
+#
+
+from copy import copy
+from dataclasses import dataclass
+from typing import Callable, cast
+
+import numpy as np
+
+from executorch.backends.nxp.backend.ir.converter.builder.model_builder import (
+    ModelBuilder,
+)
+from executorch.backends.nxp.backend.ir.converter.conversion import aten_translator
+from executorch.backends.nxp.backend.ir.converter.conversion.common import OpsList
+from executorch.backends.nxp.backend.ir.converter.tensor_utils import tensor_has_data
+from executorch.backends.nxp.backend.ir.lib.tflite.Padding import Padding
+from executorch.backends.nxp.backend.ir.tflite_generator import tflite_model
+from executorch.backends.nxp.backend.ir.tflite_generator.builtin_options import (
+    concatenation_options,
+    conv_2d_options,
+    split_options,
+)
+from torch.fx import Node
+
+
+@dataclass
+class ConvParameters:
+    stride: list[int]
+    padding: list[int]
+    dilation: list[int]
+    groups: int
+
+
+# noinspection PyPep8Naming
+def _get_IO_channels(node: Node | tflite_model.Operator) -> (int, int):
+    if isinstance(node, Node):
+        input_channels = (
+            node.args[0].meta["val"].shape[1]
+        )  # Channels of the main input.
+        output_channels = (
+            node.args[1].meta["val"].shape[0]
+        )  # Output channels of the weights.
+    else:
+        input_channels = node.tmp_inputs[0].shape[-1]  # Channels of the main input.
+        output_channels = node.tmp_inputs[1].shape[0]  # Output channels of the weights.
+
+    return input_channels, output_channels
+
+
+def group_conv_convertible_as_depthwise(node: Node | tflite_model.Operator, group: int):
+    input_channels, output_channels = _get_IO_channels(node)
+
+    return input_channels == output_channels == group
+
+
+def group_conv_convertible_into_multiple_convolutions(
+    node: Node | tflite_model.Operator, group: int
+) -> bool:
+    if group == 1:
+        return False
+
+    _, output_channels = _get_IO_channels(node)
+    if output_channels % group != 0:
+        return False  # Unable to split group Conv into separated convolutions because out_channels % group != 0.
+
+    # 10 is an empirical value. The `group` directly dictates how many branches will be created.
+    return 2 <= group <= 10
```

Comment:
Why limit the number of groups, is this a HW limit?


Is there any logging in this node converters? This will be relatively opaque if it fails since we return False for multiple different reasons